starter-code/src/viking.js

```diff
@@ -1,11 +1,95 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+    this.attack = function() {
+        return this.strength;
+    }
+    this.receiveDamage = function(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-function Viking() {}
+function Viking(name, healthArg, strengthArg) {
+
+    Soldier.call(this, healthArg, strengthArg)
+    this.name = name;
+    this.receiveDamage = function(damage) {
+        this.health -= damage;
+        if (this.health > 0){
+        return this.name + " has received " + damage + " points of damage";
+        } else {
+            return this.name + " has died in act of combat";
+        }
+    }
+
+    this.battleCry = function() {
+        return "Odin Owns You All!";
+    }
+}
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
```

Comment:
If you use the "simpler" version of oop, you don't need this two lines (31 and 32). If you add your methods with prototype, then you will need them.